src/SFA.DAS.ApplyService.Web/Services/OverallOutcomeAugmentationService.cs

```diff
@@ -0,0 +1,334 @@
+            {
+                AddPagesToSequencesFromFailedDetails(sequences, moderationFailedDetails);
+
+                var sequencesWithModerationFails = new List<AssessorSequence>();
+                BuildSequencesWithModerationFails(sequences, sequencesWithModerationFails);
+
+                var allSections = await _qnaApiClient.GetSections(model.ApplicationId);
+
+                RemoveInactiveOrEmptyPagesSectionsSequencesFromSequences(sequencesWithModerationFails, allSections);
+                AddPageTitlesToSequences(sequencesWithModerationFails);
+                AddAnswersToSequences(sequencesWithModerationFails, allSections);
+                AddQuestionsToSequences(sequencesWithModerationFails, allSections);
+                //TESTING COMPLETE TO THIS POINT
+                AddSequenceTitlesToSequences(sequencesWithModerationFails);
+
+                model.Sequences = sequencesWithModerationFails;
+                model.PagesWithGuidance =
+                    GatherGuidancePagesForSequenceQuestions(sequencesWithModerationFails, allSections);
+            }
+        }
+
+        private void AddSequenceTitlesToSequences(List<AssessorSequence> sequencesWithModerationFails)
+        {
+            foreach (var sequence in sequencesWithModerationFails)
+            {
+                sequence.SequenceTitle =
+                    _assessorLookupService.GetTitleForSequence(sequence.SequenceNumber);
+            }
+        }
+
+        private void AddQuestionsToSequences(List<AssessorSequence> sequencesWithModerationFails,
+            IEnumerable<ApplicationSection> allSections)
+        {
+            foreach (var sequence in sequencesWithModerationFails)
+            {
+                foreach (var section in sequence.Sections)
+                {
+                    foreach (var page in section.Pages)
+                    {
+                        var selectedSection = allSections.FirstOrDefault(x =>
+                            x.SequenceId == section.SequenceNumber && x.SectionId == section.SectionNumber);
+
+                        var pageDetails =
+                            selectedSection?.QnAData?.Pages?.SingleOrDefault(p => p.PageId == page.PageId);
+
+                        if (pageDetails != null && pageDetails.Active)
+                        {
+                            page.Questions = GatherAndTypePageQuestions(pageDetails);
+                        }
+                    }
+                }
+            }
+        }
+
+        private List<PageWithGuidance> GatherGuidancePagesForSequenceQuestions(
+            List<AssessorSequence> sequencesWithModerationFails, IEnumerable<ApplicationSection> allSections)
+        {
+            var guidancePages = new List<PageWithGuidance>();
+            foreach (var sequence in sequencesWithModerationFails)
+            {
+                foreach (var section in sequence.Sections)
+                {
+                    foreach (var page in section.Pages)
```

Comment:
Same as above: Could this be flattened using LINQ?

How about flattened collection being passed into function?

Reply:
ok